tensorflow/python/kernel_tests/broadcast_to_ops_test.py

```diff
@@ -81,5 +82,24 @@ def testBroadcastToShapeTypeAndInference(self):
         # check shape inference when shape input is constant
         self.assertAllEqual(shape, v_np.shape)
 
+  def testGradient(self):
```

Comment:
Can you add a few more tests?

Specifically, the gradient for broadcasting from a scalar to a vector (increasing rank). Also the case of broadcasting when the rank does not increase (say from shape [2, 1, 3] to shape [2, 5, 3]). Also one case which covers both (broadcasting from shape [2, 1] to shape [5, 2, 3]).

Reply:
Sure. Is it good to register gradient for C++ side as well?

Reply:
By the way, do we need to create a new file for gradient, say broadcast_to_ops_grad_test.py ?

Reply:
No need to make a new file or register C++ gradients.